src/ipc/ipc.c

```diff
@@ -211,7 +210,7 @@ int ipc_comp_new(struct ipc *ipc, struct sof_ipc_comp *comp)
 
 	platform_shared_commit(icd, sizeof(*icd));
 
-	return ret;
+	return 0;
```

Comment:
in fact last two patches and this could be in a separate PR for fixes as they are dont have to block on WIP status.